### ISSUE TITLE
gnt-cluster upgrade fails at ssh step when not using agent

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1121]
Originally reported of Google Code with ID 1065.

```
<b>What steps will reproduce the problem?</b>
1. create cluster in 2.12
2. log into master not using an ssh agent
3. upgrade to 2.13

<b>What is the expected output? What do you see instead?</b>

It fails at the gnt-cluster renew-crypto --new-ssh-keys step.




```
Originally added on 2015-04-15 15:34:46 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-04-15 15:38:13 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed with:

commit ee51a40b91fa5435376ea50f4480588508f0e3a4
Author: Helga Velroyen <helgav@google.com>
Date:   Thu Apr 16 10:16:14 2015 +0200

    Only delete old node keys in one-key-setup
    
    From Ganeti 2.12 to 2.13, we switched from a one-ssh-key setup
    to a setup with individual SSH keys. Unfortunately, the current
    implementation of renew-crypto deleted the master key when it
    intended to delete a non-master node's key if it was run in a
    one-ssh-key setup. This patch fixes it by fetching the key of
    the node first and only delete it, if it differs from the
    master key.
    
    This fixes bug 1065.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2015-04-16 13:15:13 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:37:03 +0000 UTC.

Changed State: Released
